sigmastate/src/main/scala/sigmastate/serialization/ValUseSerializer.scala

```diff
@@ -12,7 +12,9 @@ case class ValUseSerializer(cons: (Int, SType) => Value[SType]) extends ValueSer
   }
 
   override def parse(r: SigmaByteReader): Value[SType] = {
-    val id = r.getUIntExact
+    val id = r.getUInt().toInt
+    // Note, when id < 0 as a result of Int overflow, the r.valDefTypeStore(id) won't throw
+    // but this will likely fail elsewhere
```

Comment:
Such comment is confusing, "likely fail elsewhere" does not really says anything, please specify how a transaction where a script has negative id encoded will fail. It it will fail, no need to revert getUIntExact

Reply:
Added comment.